### PR TITLE
[FAT-1533] mod-inn-reach: Implement API Karate tests - Location by Id

### PR DESCRIPTION
**Pull request by aleksandr2110**
Purpose
https://issues.folio.org/browse/FAT-1533
Karate implementation is created for the following endpoints
GET /locations/{id}
PUT /locations/{id}
DELETE /locations/{id}
Acceptence criteria
API Tests are created at  https://github.com/folio-org/folio-integration-tests
Both positive and negative cases covered
Test are shown as passed in Cucumber report.
Test run results are visible in TestRail: https://foliotest.testrail.io/index.php?/suites/overview/19
Automated regular runs are executed at https://jenkins-aws.indexdata.com/job/FOLIO_Reference_Builds/job/folio-integration/

### CODE REVIEW

**Comment by aleksandr2110** [comment]
image

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 1]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
+     {
+      "code": "scdes",
+      "description": "Steelcase Design Library"
+     }
+    """
+    When method POST
+    Then status 201
+    * def locationId1 = $.id
+
+    * print 'Get central server by id 1'
```
Comment (dmtkachenko) [inline mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:35]:
```suggestion
    * print 'Get inn reach location by id 1'
```

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 2]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
+     {
+      "code": "plgen",
+      "description": "GVSU Pew Library General Collection"
+     }
+    """
+    When method POST
+    Then status 201
+    * def locationId2 = $.id
+
+    * print 'Get central server by id 2'
```
Comment (dmtkachenko) [inline mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:57]:
```suggestion
    * print 'Get inn reach location by id 2'
```

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 3]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
     * print 'Update location'
+    Given path 'inn-reach/locations'
+    And request
+    """
+     {
+      "code": "ongen",
+      "description": "GVSU Steelcase Library General Collection"
+     }
+    """
+    When method POST
```
Comment (dmtkachenko) [inline mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:92]: POST method for updating?
should be PUT and Response 204 according to https://s3.amazonaws.com/foliodocs/api/mod-inn-reach/s/settings.html#operation/updateLocation

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 4]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
+    Given path '/inn-reach/locations', locationId
+    And request locationResponseBefore
+    When method PUT
+    Then status 204
+
+    Given path '/inn-reach/locations', locationId
+    When method GET
+    Then status 200
+    * def locationResponseAfter = $
+    * set locationResponseAfter.code = "xxgen"
```
Comment (dmtkachenko) [inline mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:112]: not sure what you're testing here. shouldn't there be verification that updated values returned?

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 5]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
+    * set locationResponseAfter.code = "WV New Library General Collection"
```
Comment (dmtkachenko) [inline mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:113]: `code` or `description`?

Comment (aleksandr2110) [reply mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:113]: description

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature (hunk 2 above)
Comment (aleksandr2110) [reply mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:57]: Fixed

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature (hunk 1 above)
Comment (aleksandr2110) [reply mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:35]: Fixed

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature [hunk 6]
```diff
@@ -13,23 +15,125 @@ Feature: Inn reach location
...
+    Given path '/inn-reach/locations', locationId
+    When method GET
+    Then status 200
+    * def locationResponseBefore = $
+    * set locationResponseBefore.code = "eugen"
+    * set locationResponseBefore.description = "EU Simple Library General Collection"
+
+    Given path '/inn-reach/locations', locationId
+    When method DELETE
+    Then status 204
```
[the remark this answers is not in the document]
Comment (aleksandr2110) [reply mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:139]: Fixed

File: mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature (hunk 4 above)
Comment (aleksandr2110) [reply mod-inn-reach/src/main/resources/volaris/features/inn-reach-location.feature:112]: Yes Here should be 
    And match locationResponseAfter.code == "plgen"
    And match locationResponseAfter.description == "GVSU Pew Library General Collection"

**Pull request merged**